src/OpenTelemetry.Exporter.Geneva/Internal/EventHeader/EventHeaderLogExporter.cs

```diff
@@ -0,0 +1,352 @@
+// Copyright The OpenTelemetry Authors
+// SPDX-License-Identifier: Apache-2.0
+
+#if NET
+
+using System.Globalization;
+using Microsoft.LinuxTracepoints.Provider;
+using OpenTelemetry.Exporter.Geneva.Tld;
+using OpenTelemetry.Exporter.Geneva.Transports;
+using OpenTelemetry.Internal;
+using OpenTelemetry.Logs;
+
+namespace OpenTelemetry.Exporter.Geneva.EventHeader;
+
+internal class EventHeaderLogExporter : TldLogCommon, IDisposable
+{
+    internal const int StringLengthLimit = (1 << 14) - 1; // 16 * 1024 - 1 = 16383
+
+    private static readonly ThreadLocal<EventHeaderDynamicBuilder> EventBuilder = new();
+    private static readonly Action<LogRecordScope, EventHeaderLogExporter> ProcessScopeForIndividualColumnsAction = OnProcessScopeForIndividualColumns;
+
+    private readonly ValueTuple<byte[], byte[]>? repeatedPartAFields;
+
+#pragma warning disable CA2213 // Disposable fields should be disposed: it is managed and disposed in the Dispose method of EventHeaderDynamicProvider
+    private readonly EventHeaderDynamicTracepoint logsTracepoint;
+#pragma warning restore CA2213 // Disposable fields should be disposed
```

Comment:
This is an unusual pattern. I think the comment could explain this better to help others seeing this for the first time:

```suggestion
#pragma warning disable CA2213 // Disposable fields should be disposed: logsTracepoint is a local reference set in the ctor. It's lifecycle is managed by UnixUserEventsDataTransport and is disposed in EventHeaderDynamicProvider.Dispose
    private readonly EventHeaderDynamicTracepoint logsTracepoint;
#pragma warning restore CA2213 // Disposable fields should be disposed
```